memory_core/memory_core_magma.py

```diff
@@ -105,36 +129,79 @@ def __init__(self, data_width, data_depth):
                 self.wire(or_gate.ports[f"I{idx}"],
                           core_feature.ports.config[t_name])
             or_gates[t_name] = or_gate
+
         self.wire(or_gates["config_addr"].ports.O,
                   self.underlying.ports.config_addr[24:32])
         self.wire(or_gates["config_data"].ports.O,
                   self.underlying.ports.config_data)
+
         # only the first one has config_en
-        self.wire(self.__features[0].ports.config.write[0],
-                  self.underlying.ports.config_en)
+#        self.wire(self.__features[0].ports.config.write[0],
+#                  self.underlying.ports.config_en)
 
         # read data out
         for idx, core_feature in enumerate(self.__features):
-            self.add_port(f"read_config_data_{idx}",
-                          magma.Out(magma.Bits[32]))
-            # port aliasing
-            core_feature.ports["read_config_data"] = \
-                self.ports[f"read_config_data_{idx}"]
+            if(idx > 0):
+                self.add_port(f"read_config_data_{idx}",
+                              magma.Out(magma.Bits[32]))
+                # port aliasing
+                core_feature.ports["read_config_data"] = \
+                    self.ports[f"read_config_data_{idx}"]
+
         # MEM config
-        self.wire(self.ports.read_config_data_0,
-                  self.underlying.ports.read_data)
+        # self.wire(self.ports.read_config_data,
+        #          self.underlying.ports.read_config_data)
+
+        configuration_add = []
+        configuration_add.append(("stencil_width", 32))
```

Comment:
Can you write it as list literals? Something like
```Python
configurations = [
     ("stencil_width", 32),
     ("read_mode", 1)
]
```

Reply:
yep, done